### DISCUSSION TITLE
Store Adding Rather Than Replacing Data

### DISCUSSION THREAD

**Discussion by darrylmorley** [ai/nanostores#159] [Q&A]
Hi,

As in the title. I have set up a store (items) which I am using to store an API response. When the API is called again the data is added to the existing data, rather than replacing the existing data as I would expect. Does a store need to be cleared first or.. why would this be?

```
if (res.ok) items.set(await res.json());
```

**Comment by ai** [comment]
`items.set(await res.json())` should work. Can you show a full example of the store?

**Reply by darrylmorley** [reply]
I have this:
```
import { atom } from "nanostores";

export const items = atom<any[]>([]);
```

**Reply by ai** [reply]
Show me the code which loads data

**Reply by darrylmorley** [reply]
I have this function:
```js
  const fetchItems = async () => {
    let lines = orderLines.get();
    let ids = lines.map((line) => line.itemID);

    if (ids.length === lines.length) {
      const res = await fetch(`/api/items/${encodeURIComponent(ids)}.json`);
      if (res.ok) items.set(await res.json());
    }
  };
```

**Reply by ai** [reply]
Very strange. It should just work.

How do you test it? Do you check it only in UI (maybe it is a bug in your UI library, try `console.log(items.get())`)?

Also debug what `await res.json()` return. Maybe it is a server bug?

**Reply by darrylmorley** [reply]
Yes, just in UI & console.log. Will do, I can see the array getting larger in console.log.

Thanks for the help, I can see that it's an issue when pulling the data!
